multiversx_sdk_rust_contract_builder/builder.py

```diff
@@ -1,106 +1,105 @@
+    MAX_OUTPUT_ARTIFACTS_ARCHIVE_SIZE, MAX_PACKAGED_SOURCE_CODE_SIZE,
+    OLD_CONTRACT_CONFIG_FILENAME)
+from multiversx_sdk_rust_contract_builder.errors import ErrKnown
 from multiversx_sdk_rust_contract_builder.filesystem import (
-    archive_directory, find_file_in_folder)
+    archive_folder, find_file_in_folder)
 from multiversx_sdk_rust_contract_builder.packaged_source_code import \
     PackagedSourceCode
-from multiversx_sdk_rust_contract_builder.source_code import \
-    is_source_code_file
 from multiversx_sdk_rust_contract_builder.wabt import generate_wabt_artifacts
 
 
 def build_project(
-        project_path: Path,
-        parent_output_directory: Path,
-        specific_contract: Union[Path, None],
+        project_folder: Path,
+        parent_output_folder: Path,
+        specific_contract: Optional[str],
         cargo_target_dir: Path,
-        no_wasm_opt: bool) -> BuildOutcome:
-    project_path = project_path.expanduser().resolve()
-    parent_output_directory = parent_output_directory.expanduser().resolve()
+        no_wasm_opt: bool,
+        context: str) -> BuildOutcome:
+    project_folder = project_folder.expanduser().resolve()
+    parent_output_folder = parent_output_folder.expanduser().resolve()
     cargo_target_dir = cargo_target_dir.expanduser().resolve()
 
-    outcome = BuildOutcome()
-    contracts_directories = get_contracts_directories(project_path)
+    outcome = BuildOutcome(context)
+    contracts_folders = get_contracts_folders(project_folder)
 
     # We copy the whole project folder to the build path, to ensure that all local dependencies are available.
-    project_within_build_directory = copy_project_directory_to_build_directory(project_path)
+    project_within_build_folder = copy_project_folder_to_build_folder(project_folder)
 
-    for contract_directory in sorted(contracts_directories):
-        contract_name, contract_version = get_contract_name_and_version(contract_directory)
+    for contract_folder in sorted(contracts_folders):
+        contract_name, contract_version = get_contract_name_and_version(contract_folder)
         logging.info(f"Contract = {contract_name}, version = {contract_version}")
 
-        output_subdirectory = parent_output_directory / f"{contract_name}"
-        output_subdirectory.mkdir(parents=True, exist_ok=True)
-
-        relative_contract_directory = contract_directory.relative_to(project_path)
-        build_directory = project_within_build_directory / relative_contract_directory
-
         if specific_contract and contract_name != specific_contract:
             logging.info(f"Skipping {contract_name}.")
             continue
 
-        # Clean directory - useful if it contains externally-generated build artifacts
-        clean_contract(build_directory)
-        build_contract(build_directory, output_subdirectory, cargo_target_dir, no_wasm_opt)
+        output_subfolder = parent_output_folder / f"{contract_name}"
```

Comment:
We've just moved the check `if specific_contract and contract_name != specific_contract` above.